lib/blocks.php

```diff
@@ -504,3 +504,94 @@ function gutenberg_migrate_old_typography_shape( $metadata ) {
 if ( ! function_exists( 'wp_migrate_old_typography_shape' ) ) {
 	add_filter( 'block_type_metadata', 'gutenberg_migrate_old_typography_shape' );
 }
+
+/**
+ * Enqueue a stylesheet for a specific block.
+ *
+ * If the theme has opted-in to separate-styles loading,
+ * then the stylesheet will be enqueued on-render,
+ * otherwise when the block inits.
+ *
+ * @param string $block_name The block-name, including namespace.
+ * @param array  $args       An array of arguments [handle,src,deps,ver,media].
+ * @param bool   $register   Whether the stylesheet should be registered.
+ *                           If set to false, then assume it has already been registered,
+ *                           and only enqueue it.
+ *
+ * @return void
+ */
+function gutenberg_enqueue_block_style( $block_name, $args, $register = true ) {
+	$args = wp_parse_args(
+		$args,
+		array(
+			'handle' => '',
+			'src'    => '',
+			'deps'   => array(),
+			'ver'    => false,
+			'media'  => 'all',
+		)
+	);
+
+	/**
+	 * Callback function to register and enqueue styles.
+	 *
+	 * @param string $content When the callback is used for the render_block filter,
+	 *                        the content needs to be returned so the function parameter
+	 *                        is to ensure the content exists.
+	 *
+	 * @return string
+	 */
+	$callback = function( $content ) use ( $args, $register ) {
+		// Register the stylesheet.
+		if ( $register ) {
+			wp_register_style( $args['handle'], $args['src'], $args['deps'], $args['ver'], $args['media'] );
+		}
+
+		// Add `path` data if provided.
+		if ( isset( $args['path'] ) ) {
+			wp_style_add_data( $args['handle'], 'path', $args['path'] );
+		}
+
+		// Enqueue the stylesheet.
+		wp_enqueue_style( $args['handle'] );
+
+		return $content;
+	};
+
+	$hook = did_action( 'wp_enqueue_scripts' ) ? 'wp_footer' : 'wp_enqueue_scripts';
+	if ( wp_should_load_separate_core_block_assets() ) {
+		$hook = "render_block_$block_name";
+	}
+
+	// Enqueue assets in the frontend.
+	add_filter( $hook, $callback );
+
+	// Enqueue assets in the editor.
+	add_action( 'enqueue_block_editor_assets', $callback );
```

Comment:
This won't work in the template editor. Styles must either be added to the block registration or added as editor styles. `enqueue_block_editor_assets` is for assets for the block editor UI, not for the editor content.

Reply:
> This won't work in the template editor. Styles must either be added to the block registration or added as editor styles. `enqueue_block_editor_assets` is for assets for the block editor UI, not for the editor content.

Thank you for the insights!
Unfortunately using [`add_editor_style()`](https://developer.wordpress.org/reference/functions/add_editor_style/) is not an option 'cause that function accepts stylesheets with a path relative to the theme root, so that would make it impossible for plugins to use the function properly.
I changed it to use the `enqueue_block_assets` hook and that seems to work fine on my test. Would that hook cause any issues?

Reply:
Honestly, I didn't know about `enqueue_block_assets`! I'll have a look.